src/snac/data/AbstractData.php

```diff
@@ -240,7 +244,7 @@ protected function diffArray($first, $second, $strict = true) {
         foreach ($first as $data) {
             $seen = false;
             foreach ($second as $k => $odata) {
-                if ($data != null && $data->equals($odata, $strict)
+                if ($data != null && $data->equals($odata, $strict, $checkSubcomponents)
```

Comment:
We may need to think more about section.  Since the diffArray response shouldn't drop any data, we may want to check conditions:
```
if (should check subcomponents)
    push one copy onto the intersection array // line252, pushing the $data copy
else //shouldn't check subcomponents
    create $combined
    $combined->combine($data)
    $combined->combine($odata)
    push $combined onto the intersection
```

Reply:
After our discussion, I'll remove check subcomponents from diffArray for the time being, until we know how PHP will deal with AbstractData's children for combine().